Added support for Minimal Upgrade

vik-unimus opened the pull request.

vik-unimus committed:
Added support for Rocky Linux + Docker image source

vik-unimus committed:
Added support for Rocky Linux

vik-unimus committed:
Added support for Rocky Linux

vik-unimus committed:
Added support for minimal upgrade

Added support for "-m" argument and additional conditions to prevent certain functions to do proceed if minimal upgrade is chosen.

vik-unimus committed:
Added support for minimal upgrade

vik-unimus committed:
Added support for minimal upgrade

vik-unimus committed:
Fixed some typos

vik-unimus committed:
Removed Rocky Linux support

Added via a separate branch.

vik-unimus committed:
Removed Rocky Linux support

Added in a separate branch.

vik-unimus committed:
Removed Rocky Linux support

Added in a separate branch.

vik-unimus committed:
Removed installation checker

Will be replaced by a separate function in common functions.

vik-unimus committed:
Removed installation checker

Will be replaced by a separate function in common functions.

vik-unimus committed:
Added function to check existing installation

tomaskir commented on src/common/functions/functions.sh:
```diff
@@ -77,6 +80,21 @@ function check_root {
 
+function check_existing_installation {
+  # check existing installation if -m argument (minimal upgrade) is detected
+  for i in ${run_args[@]}; do
```
No need to checks args, we can just check if `${minimal} == 1`

tomaskir commented on src/common/functions/functions.sh:
```diff
@@ -77,6 +80,21 @@ function check_root {
+  # check existing installation if -m argument (minimal upgrade) is detected
+  for i in ${run_args[@]}; do
+    if [[ $i == '-m' ]]; then
+      if [ -f "${binary_path}" ]; then
```
we should be able to use `[[ ! -f "${binary_path}" ]];`, and avoid needing a `break;`

tomaskir commented on src/common/functions/functions.sh:
```diff
@@ -133,22 +151,35 @@ function get_per_os_parameters {
+	  echo;
+	  echo "If you are upgrading from a previous version, your current ${product_name} service will be stopped and restarted automatically.";
+	  echo;
+	  echo 'If you experience any issues with this installer, or have any questions, please contact us.';
```
remove text duplication, move under `if` block

tomaskir commented on src/common/functions/functions.sh:
```diff
@@ -133,22 +151,35 @@ function get_per_os_parameters {
-
+  if [[ ${minimal} == 0 ]]; then
+	  echo;
+	  echo "Welcome to the ${product_name} installer!";
```
remove text duplication, move above `if` block

vik-unimus committed:
Updated with requested changes

Loop in the check_existing_installation function removed completely, replaced with two if conditions providing the same functionality. Installer text updated to minimize text duplication and if condition updated to be shorter and to the point.

vik-unimus committed:
Updated main function, removed unneccesary clutter

Installer types separated into separate functions. Removed all if statements at the beginning of the functions we didn't want to run when minimal upgrade is detected.
